workshop/content/20-typescript/70-advanced-topics/100-construct-testing/1000-assertion-test.md

```diff
@@ -0,0 +1,326 @@
+CDK will calculate a hash to append to the end of the name of the constructs, so we will just use a 
+dummy value for now. Once we run the test it will fail and show us the expected value.
+
+Create a new test in `hitcounter.test.ts` with the below code:
+
+```typescript
+test('Lambda Has Environment Variables', () => {
+  const stack = new cdk.Stack();
+  // WHEN
+  new HitCounter(stack, 'MyTestConstruct', {
+    downstream:  new lambda.Function(stack, 'TestFunction', {
+      runtime: lambda.Runtime.NODEJS_10_X,
+      handler: 'lambda.handler',
+      code: lambda.Code.inline('test')
+    })
+  });
+  // THEN
+  expectCDK(stack).to(haveResource("AWS::Lambda::Function", {
+    Environment: {
+      Variables: {
+        DOWNSTREAM_FUNCTION_NAME: "TestFunction",
+        HITS_TABLE_NAME: "MyTestConstructHits"
+      }
+    }
+  }));
+})
+```
+
+Save the file and run the test again.
+
+```bash
+$ npm run build && npx jest
+```
+
+This time the test should fail and you should be able to grab the correct value for the
+variables from the expected output.
+
+{{<highlight bash "hl_lines=16-21">}}
+$ npx jest
+
+> cdk-workshop@0.1.0 test /home/aws-cdk-intro-workshop
+> jest
+
+ FAIL  test/hitcounter.test.ts
+  ✓ DynamoDB Table Created (184ms)
+  ✕ Lambda Has Environment Variables (53ms)
+  ✓ read capacity can be configured (9ms)
+
+  ● Lambda Has Environment Variables
+  ...
+  ...
+  "Environment": {
+              "Variables": {
+                "DOWNSTREAM_FUNCTION_NAME": {
+                  "Ref": "TestFunctionXXXXX"
+                },
+                "HITS_TABLE_NAME": {
+                  "Ref": "MyTestConstructHitsXXXXX"
+                }
+              }
+            }
+          },
+          "DependsOn": [
+            "MyTestConstructHitCounterHandlerServiceRoleDefaultPolicyXXXXX",
+            "MyTestConstructHitCounterHandlerServiceRoleXXXXX"
+          ]
+        }
+
+      37 |   });
+      38 |   // THEN
+    > 39 |   expectCDK(stack).to(haveResource("AWS::Lambda::Function", {
+         |                    ^
+      40 |     Environment: {
+      41 |       Variables: {
+      42 |         DOWNSTREAM_FUNCTION_NAME: {"Ref": "TestFunction"},
+
+      ...
+      ...
+
+Test Suites: 1 failed, 1 total
+Tests:       1 failed, 2 passed, 3 total
+Snapshots:   0 total
+Time:        3.686s
+{{</highlight>}}
+
+Grab the real values for the environment variables and update your test
+
+{{<highlight ts "hl_lines=15-16">}}
+test('Lambda Has Environment Variables', () => {
+  const stack = new cdk.Stack();
+  // WHEN
+  new HitCounter(stack, 'MyTestConstruct', {
+    downstream:  new lambda.Function(stack, 'TestFunction', {
+      runtime: lambda.Runtime.NODEJS_10_X,
+      handler: 'lambda.handler',
+      code: lambda.Code.inline('test')
+    })
+  });
+  // THEN
+  expectCDK(stack).to(haveResource("AWS::Lambda::Function", {
+    Environment: {
+      Variables: {
+        DOWNSTREAM_FUNCTION_NAME: {"Ref": [VALUE_GOES_HERE]},
```

Comment:
@corymhall 

Could you please describe, how it's supposed to retrieve/substitute a real value instead of ```{"Ref": [VALUE_GOES_HERE]}```?